src/main/kotlin/com/hedvig/accountservice/schedulers/domain/MemberScheduler.kt

```diff
@@ -1,561 +1,544 @@
-    val transition = isAllowedToTransition(nextState)
-    if (transition.outcome != SchedulerStateTransitionOutcome.ALLOWED_TRANSITION) {
-      apply(BlacklistSchedulerDeniedEvent(command.memberId, currentState, nextState))
-      logger.error { "Blacklist DENIED for memberId=${command.memberId} (Message: ${transition.message})" }
-      return transition
-    }
-    apply(SchedulerBlacklistedEvent(command.memberId, nextState))
-    logger.info { "Scheduler BLACKLISTED for memberId=${command.memberId}" }
-    return transition
-  }
-
-  @EventSourcingHandler
-  fun on(event: SchedulerBlacklistedEvent) {
-    currentState = event.nextState
-  }
-
-  @CommandHandler
-  fun handle(command: RevertBlacklistSchedulerCommand): SchedulerStateTransition {
-    val nextState = SchedulerState(
-      memberId = command.memberId,
-      status = ChargeStatus.INITIATED,
-      changedBy = command.revertedBy
-    )
-    if (currentState.status != ChargeStatus.BLACKLISTED) {
-      logger.error("Tried to revert a Blacklisted state but the status is ${currentState.status} [MemberId: ${currentState.memberId}]")
-      return SchedulerStateTransition(SchedulerStateTransitionOutcome.FORBIDDEN_STATUS_TRANSITION)
-    }
-
-    apply(SchedulerBlacklistRevertedEvent(command.memberId, nextState))
-    logger.info { "Scheduler INITIATED after blacklist reverted for memberId=${command.memberId}" }
-    return SchedulerStateTransition(SchedulerStateTransitionOutcome.ALLOWED_TRANSITION)
-  }
-
-  @EventSourcingHandler
-  fun on(event: SchedulerBlacklistRevertedEvent) {
-    currentState = event.nextState
-  }
-
-  @CommandHandler
-  fun handle(command: DangerouslyResetCommand): SchedulerStateTransition {
-    if (currentState.status != ChargeStatus.SUBSCRIPTION_SCHEDULED_AND_WAITING_FOR_APPROVAL) {
-      return SchedulerStateTransition(
-        outcome = SchedulerStateTransitionOutcome.FORBIDDEN_STATUS_TRANSITION,
-        message = "Dangerously reset DENIED for member=${command.memberId} (Message: Scheduler in the state ${currentState.status} cannot be reset)"
-      )
-    }
-    val nextState = SchedulerState(
-      memberId = command.memberId,
-      status = ChargeStatus.INITIATED,
-      changedBy = command.resetBy
-    )
-    apply(SchedulerDangerouslyResetEvent(command.memberId, nextState))
-    logger.info { "Scheduler DANGEROUSLY reset for member=${command.memberId}" }
-    return SchedulerStateTransition(SchedulerStateTransitionOutcome.ALLOWED_TRANSITION)
-  }
-
-  @EventSourcingHandler
-  fun on(event: SchedulerDangerouslyResetEvent) {
-    currentState = event.nextState
-  }
-}
\ No newline at end of file
+        apply(SchedulerInitiatedEvent(command.memberId, nextState))
+        logger.info { "Scheduler INITIATED for memberId=${command.memberId}" }
+    }
+
+    @EventSourcingHandler
+    fun on(event: SchedulerInitiatedEvent, @Timestamp timestamp: Instant) {
+        currentState = event.nextState
+        initiationDate = timestamp.toLocalDate()
```

Comment:
I think we should have this more explicit for future work if we start operating in time zones.
```suggestion
        initiationDate = timestamp.toStockholmLocalDate()
```